src/BaselineUtils.cc

```diff
@@ -1037,8 +1129,16 @@ symmetrizeFootprint(
         if (dxlo <= dxhi) {
             LOGL_DEBUG(_log, "Adding span fwd %i, [%i, %i],  back %i, [%i, %i]",
                        fy, cx+dxlo, cx+dxhi, by, cx-dxhi, cx-dxlo);
-            sfoot->addSpan(fy, cx + dxlo, cx + dxhi);
-            sfoot->addSpan(by, cx - dxhi, cx - dxlo);
+            if (cx + dxlo <= cx + dxhi) {
+                tmpSpans.push_back(geom::Span(fy, cx + dxlo, cx + dxhi));
+            } else {
+                tmpSpans.push_back(geom::Span(fy, cx + dxhi, cx + dxlo));
+            }
+            if (cx - dxlo <= cx - dxhi) {
+                tmpSpans.push_back(geom::Span(by, cx - dxlo, cx - dxhi));
+            } else {
+                tmpSpans.push_back(geom::Span(by, cx - dxhi, cx - dxlo));
+            }
```

Comment:
See my previous comment for why this if..else block also seems unnecessary to me.